### PR TITLE
Firewall: Add content type information

### PR DESCRIPTION
**Pull request by pedrosousa**
Added content type to Firewall topics.
This information is only used for internal purposes -- it has no effect in the rendered docs.

Also includes:
- Fixed broken links
- Minor formatting/grammar changes

### CODE REVIEW

File: products/firewall/src/content/api/cf-filters/index.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ---
+pcx-content-type: reference
```
Comment (pedrosousa) [inline products/firewall/src/content/api/cf-filters/index.md:2]: Could also be a `landing page`, but it has some relevant information.

File: products/firewall/src/content/api/cf-lists/index.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ---
+pcx-content-type: reference
```
Comment (pedrosousa) [inline products/firewall/src/content/api/cf-lists/index.md]: Could also be a `landing page`.

File: products/firewall/src/content/cf-dashboard/expression-preview-editor.md [hunk 3]
```diff
@@ -1,18 +1,19 @@
 ---
+pcx-content-type: how to
```
Comment (pedrosousa) [inline products/firewall/src/content/cf-dashboard/expression-preview-editor.md]: Is it a `how to`?

File: products/firewall/src/content/cf-rulesets/common-use-cases/deploy-cmr-joomla-only.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Enable only Joomla rules
+pcx-content-type: configuration
```
Comment (pedrosousa) [inline products/firewall/src/content/cf-rulesets/common-use-cases/deploy-cmr-joomla-only.md:3]: I considered these recipes/common use cases as `configuration`.

File: products/firewall/src/content/cf-rulesets/custom-rules/rate-limiting/manage-api.md [hunk 5]
```diff
@@ -1,4 +1,5 @@
 ---
+pcx-content-type: how to
```
Comment (pedrosousa) [inline products/firewall/src/content/cf-rulesets/custom-rules/rate-limiting/manage-api.md]: Not sure about these... they're actionable, but they don't have numbered steps. To me, they're still a `how to`.

File: products/firewall/src/content/cf-rulesets/custom-rulesets/add-rules-ruleset.md [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Add rules to a custom ruleset
+pcx-content-type: how to
```
Comment (pedrosousa) [inline products/firewall/src/content/cf-rulesets/custom-rulesets/add-rules-ruleset.md]: Another "how to" without numbered steps.

File: products/firewall/src/content/cf-dashboard/expression-preview-editor.md (hunk 3 above)
Comment (kodster28) [reply products/firewall/src/content/cf-dashboard/expression-preview-editor.md]: I'd agree with that, yeah.

File: products/firewall/src/content/cf-rulesets/common-use-cases/deploy-cmr-joomla-only.md (hunk 4 above)
Comment (kodster28) [reply products/firewall/src/content/cf-rulesets/common-use-cases/deploy-cmr-joomla-only.md:3]: I'd agree with that. We might want to add some more info around the differences between `tutorials` and `configuration` though, just for clarity.

File: products/firewall/src/content/cf-rulesets/custom-rules/rate-limiting/manage-api.md (hunk 5 above)
Comment (kodster28) [reply products/firewall/src/content/cf-rulesets/custom-rules/rate-limiting/manage-api.md]: I'd agree with that.

File: products/firewall/src/content/api/cf-lists/index.md (hunk 2 above)
Comment (kodster28) [reply products/firewall/src/content/api/cf-lists/index.md]: This one feels like more of a `navigation` page to me, but could also see it as a `concept` page.

File: products/firewall/src/content/api/cf-filters/index.md (hunk 1 above)
Comment (kodster28) [reply products/firewall/src/content/api/cf-filters/index.md:2]: Actually, backtracking a bit on what I'm saying here.

On one level, I could see it being a `reference` or `navigation` page.

But again, I also have [examples](https://developers.cloudflare.com/page-shield/script-monitor) similar to this where I'm definitely going to be calling them `landing pages`.

Maybe that's worth a larger discussion? Of how we treat a feature that falls w/in a larger product?

**Review by kodster28 — APPROVED** [review]

File: products/firewall/src/content/api/cf-lists/index.md (hunk 2 above)
Comment (pedrosousa) [reply products/firewall/src/content/api/cf-lists/index.md]: Considering what you said before about `landing page`s, it's probably a `navigation` page, yes.

File: products/firewall/src/content/api/cf-filters/index.md (hunk 1 above)
Comment (crwaters16) [reply products/firewall/src/content/api/cf-filters/index.md:2]: I think it's your last point, @kodster28 - we should have a discussion around how we treat features/components within a larger product. I have examples of this type of page for logpush/logpull, for example.

File: products/firewall/src/content/api/cf-lists/index.md [hunk 7]
```diff
@@ -1,4 +1,5 @@
 ---
+pcx-content-type: navigation
```
Comment (crwaters16) [inline products/firewall/src/content/api/cf-lists/index.md:2]: Part of me expects all of these API pages to be the same content type - worth discussing in content critique

File: products/firewall/src/content/cf-dashboard/expression-preview-editor.md (hunk 3 above)
Comment (crwaters16) [reply products/firewall/src/content/cf-dashboard/expression-preview-editor.md]: Agreed - though I do think this is an example of a page with multiple content types on it :)

File: products/firewall/src/content/cf-dashboard/rule-preview.md [hunk 8]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Preview rules
+pcx-content-type: reference
```
Comment (crwaters16) [inline products/firewall/src/content/cf-dashboard/rule-preview.md]: I think you have both reference and how-to content in here

File: products/firewall/src/content/cf-rulesets/common-use-cases/deploy-cmr-joomla-only.md (hunk 4 above)
Comment (crwaters16) [reply products/firewall/src/content/cf-rulesets/common-use-cases/deploy-cmr-joomla-only.md:3]: I've got an empty page in the wiki for a config vs tutorial vs example, and I'd expect us to iron this out more as we define the strategy and template for each content type. This page is one that's currently a combo of both content types and would benefit from guidance on whether to become one or the other.

File: products/firewall/src/content/cf-rulesets/custom-rules/rate-limiting/manage-api.md (hunk 5 above)
Comment (crwaters16) [reply products/firewall/src/content/cf-rulesets/custom-rules/rate-limiting/manage-api.md]: Agree that they're telling you how to do something - and once there's template/strategy, something I'd expect to see evolve to include steps

File: products/firewall/src/content/cf-dashboard/rule-preview.md (hunk 8 above)
Comment (pedrosousa) [reply products/firewall/src/content/cf-dashboard/rule-preview.md]: Changing to `interim`.

File: products/firewall/src/content/cf-dashboard/rule-preview.md [hunk 9]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Preview rules
+pcx-content-type: interim
```
Comment (pedrosousa) [inline products/firewall/src/content/cf-dashboard/rule-preview.md:3]: Using "interim" for mixed content (more than one content type) in topics that need some work.

**Pull request merged**